tests/lib/nm/ipv4_test.py

```diff
@@ -16,6 +16,7 @@
 #
 
 import pytest
+import socket
```

Comment:
Why not leave it up to the ipv4 module to provide it? What is the advantage of exposing the test module to this dependency?
Same for the IPv6 test module.